<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QMainWindow" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1369</width>
    <height>851</height>
   </rect>
  </property>
  <property name="focusPolicy">
   <enum>Qt::StrongFocus</enum>
  </property>
  <property name="windowTitle">
   <string>GPS Map Construction</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QFrame" name="selectionFrame">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>260</width>
        <height>0</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBoxTraj">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>260</width>
           <height>0</height>
          </size>
         </property>
         <property name="font">
          <font>
           <family>Lucida Grande</family>
           <pointsize>14</pointsize>
          </font>
         </property>
         <property name="layoutDirection">
          <enum>Qt::LeftToRight</enum>
         </property>
         <property name="title">
          <string>Trajectories</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="trajInfo">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>No trajectories has been loaded yet.</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox">
            <property name="font">
             <font>
              <pointsize>13</pointsize>
             </font>
            </property>
            <property name="layoutDirection">
             <enum>Qt::RightToLeft</enum>
            </property>
            <property name="text">
             <string>Show interpolated</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QListView" name="trajListView">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="mapGroupBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <pointsize>14</pointsize>
          </font>
         </property>
         <property name="title">
          <string>Map</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QCheckBox" name="showMapCheckBox">
            <property name="font">
             <font>
              <pointsize>13</pointsize>
             </font>
            </property>
            <property name="layoutDirection">
             <enum>Qt::RightToLeft</enum>
            </property>
            <property name="text">
             <string>Show Map</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="mapInfo">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>No osm file has been loaded yet.</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="sampleGroupBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <pointsize>14</pointsize>
          </font>
         </property>
         <property name="title">
          <string>Sample</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QCheckBox" name="showSampleCheckBox">
            <property name="font">
             <font>
              <pointsize>13</pointsize>
             </font>
            </property>
            <property name="layoutDirection">
             <enum>Qt::RightToLeft</enum>
            </property>
            <property name="text">
             <string>Show Samples</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="sampleInfoLabel">
            <property name="text">
             <string>No samples have been drawn yet.</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_7">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QFrame" name="frame_4">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>310</width>
                 <height>64</height>
                </size>
               </property>
               <property name="frameShape">
                <enum>QFrame::NoFrame</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <property name="lineWidth">
                <number>0</number>
               </property>
               <widget class="QPushButton" name="clusterSegments">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="geometry">
                 <rect>
                  <x>207</x>
                  <y>37</y>
                  <width>81</width>
                  <height>20</height>
                 </rect>
                </property>
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Cluster</string>
                </property>
               </widget>
               <widget class="QDoubleSpinBox" name="sigmaValue">
                <property name="geometry">
                 <rect>
                  <x>48</x>
                  <y>6</y>
                  <width>111</width>
                  <height>24</height>
                 </rect>
                </property>
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="suffix">
                 <string>m/s</string>
                </property>
                <property name="minimum">
                 <double>0.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>10.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.050000000000000</double>
                </property>
                <property name="value">
                 <double>0.200000000000000</double>
                </property>
               </widget>
               <widget class="QLabel" name="label">
                <property name="geometry">
                 <rect>
                  <x>8</x>
                  <y>10</y>
                  <width>43</width>
                  <height>16</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Sigma</string>
                </property>
               </widget>
               <widget class="QLabel" name="label_7">
                <property name="geometry">
                 <rect>
                  <x>8</x>
                  <y>40</y>
                  <width>111</width>
                  <height>16</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Min Cluster Size</string>
                </property>
               </widget>
               <widget class="QSpinBox" name="minClusterSize">
                <property name="geometry">
                 <rect>
                  <x>110</x>
                  <y>35</y>
                  <width>49</width>
                  <height>24</height>
                 </rect>
                </property>
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>20</number>
                </property>
                <property name="value">
                 <number>5</number>
                </property>
               </widget>
               <widget class="QLabel" name="label_8">
                <property name="geometry">
                 <rect>
                  <x>170</x>
                  <y>10</y>
                  <width>81</width>
                  <height>16</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Show cluster</string>
                </property>
               </widget>
               <widget class="QSpinBox" name="showClusterId">
                <property name="geometry">
                 <rect>
                  <x>255</x>
                  <y>6</y>
                  <width>51</width>
                  <height>24</height>
                 </rect>
                </property>
                <property name="maximum">
                 <number>10000000</number>
                </property>
               </widget>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Plain</enum>
            </property>
            <property name="lineWidth">
             <number>0</number>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Search range:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="sampleCoverDistanceSpinBox">
               <property name="suffix">
                <string>m</string>
               </property>
               <property name="decimals">
                <number>1</number>
               </property>
               <property name="singleStep">
                <double>5.000000000000000</double>
               </property>
               <property name="value">
                <double>10.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="segmentGroupBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <pointsize>14</pointsize>
          </font>
         </property>
         <property name="title">
          <string>Segment</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QCheckBox" name="showSegmentCheckBox">
            <property name="font">
             <font>
              <pointsize>13</pointsize>
             </font>
            </property>
            <property name="layoutDirection">
             <enum>Qt::RightToLeft</enum>
            </property>
            <property name="text">
             <string>Show Segment</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="segmentInfoLabel">
            <property name="text">
             <string>No segments have been drawn yet.</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_2">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Pick segment at:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="segmentIdx">
               <property name="maximum">
                <number>10000000</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="graphGroupBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <pointsize>14</pointsize>
          </font>
         </property>
         <property name="title">
          <string>Graph</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QCheckBox" name="showGraphCheckBox">
            <property name="font">
             <font>
              <pointsize>13</pointsize>
             </font>
            </property>
            <property name="layoutDirection">
             <enum>Qt::RightToLeft</enum>
            </property>
            <property name="text">
             <string>Show Graph</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="graphInfoLabel">
            <property name="text">
             <string>No graph has been computed yet.</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="SceneWidget" name="scene_widget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1369</width>
     <height>22</height>
    </rect>
   </property>
   <property name="defaultUp">
    <bool>false</bool>
   </property>
   <property name="nativeMenuBar">
    <bool>true</bool>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSetWorkspace"/>
    <addaction name="separator"/>
    <addaction name="actionOpenTrajectories"/>
    <addaction name="actionSaveTrajectories"/>
    <addaction name="separator"/>
    <addaction name="actionOpenOsmMap"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionExtractTrajectory"/>
    <addaction name="separator"/>
    <addaction name="actionComputeDistanceGraph"/>
    <addaction name="separator"/>
    <addaction name="actionSamplePointCloud"/>
    <addaction name="actionPickSample"/>
    <addaction name="actionClearPickedSamples"/>
    <addaction name="separator"/>
    <addaction name="actionGenerateSegments"/>
    <addaction name="separator"/>
    <addaction name="actionInitializeGraph"/>
    <addaction name="actionClusterSampleSegments"/>
    <addaction name="actionInterpolateSegments"/>
    <addaction name="actionUpdateGraph"/>
   </widget>
   <widget class="QMenu" name="menuAlgorithms">
    <property name="title">
     <string>Algorithms</string>
    </property>
    <addaction name="actionCutFlows"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuAlgorithms"/>
   <addaction name="menuTools"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="autoFillBackground">
    <bool>true</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="actionSamplePointCloud"/>
   <addaction name="actionPickSample"/>
   <addaction name="actionClearPickedSamples"/>
   <addaction name="actionClusterSampleSegments"/>
   <addaction name="separator"/>
   <addaction name="actionGenerateSegments"/>
   <addaction name="separator"/>
   <addaction name="actionInitializeGraph"/>
   <addaction name="actionUpdateGraph"/>
   <addaction name="actionInterpolateSegments"/>
  </widget>
  <action name="actionSetWorkspace">
   <property name="text">
    <string>Set Workspace</string>
   </property>
  </action>
  <action name="actionToggleTrajectories">
   <property name="text">
    <string>Toggle Trajectories</string>
   </property>
  </action>
  <action name="actionAddValve">
   <property name="text">
    <string>Add Valve</string>
   </property>
  </action>
  <action name="actionOpenTrajectories">
   <property name="text">
    <string>Open Trajectories</string>
   </property>
  </action>
  <action name="actionColorizeUniform">
   <property name="text">
    <string>Colorize Uniform</string>
   </property>
  </action>
  <action name="actionColorizeSampleTime">
   <property name="text">
    <string>Colorize Sample Time</string>
   </property>
  </action>
  <action name="actionColorizeSampleOrder">
   <property name="text">
    <string>Colorize Sample Order</string>
   </property>
  </action>
  <action name="actionSaveTrajectories">
   <property name="text">
    <string>Save Trajectories</string>
   </property>
  </action>
  <action name="actionCutFlows">
   <property name="text">
    <string>Cut Flows</string>
   </property>
  </action>
  <action name="actionOpenOsmMap">
   <property name="text">
    <string>Open OSM Map</string>
   </property>
  </action>
  <action name="actionExtractTrajectory">
   <property name="text">
    <string>Extract Trajectory</string>
   </property>
  </action>
  <action name="actionSamplePointCloud">
   <property name="text">
    <string>Sample Point Cloud</string>
   </property>
  </action>
  <action name="actionGenerateSegments">
   <property name="text">
    <string>Compute Segments</string>
   </property>
  </action>
  <action name="actionPickSample">
   <property name="text">
    <string>Pick a Sample (P)</string>
   </property>
  </action>
  <action name="actionClearPickedSamples">
   <property name="text">
    <string>Clear Picked Samples (C)</string>
   </property>
  </action>
  <action name="actionToggleSegmentsAtSample">
   <property name="text">
    <string>Toggle Segments at Sample</string>
   </property>
  </action>
  <action name="actionInitializeGraph">
   <property name="text">
    <string>Initialize Graph</string>
   </property>
  </action>
  <action name="actionClusterSampleSegments">
   <property name="text">
    <string>Cluster Sample Segments</string>
   </property>
  </action>
  <action name="actionInterpolateSegments">
   <property name="text">
    <string>Interpolate Segments</string>
   </property>
  </action>
  <action name="actionUpdateGraph">
   <property name="text">
    <string>Update Graph</string>
   </property>
  </action>
  <action name="actionComputeDistanceGraph">
   <property name="text">
    <string>Compute Distance Graph</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>SceneWidget</class>
   <extends>QWidget</extends>
   <header location="global">scene_widget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="main_window.qrc"/>
 </resources>
 <connections/>
</ui>
